### ISSUE TITLE
Desktop Mac - cannot log in and do not own another device

### ISSUE THREAD

**Issue by ararename** [telegramdesktop/tdesktop#16777]
Hello erveryone. 
Since yesterday Telegram desktop for mac (not sure about the version as i already tried to update telegram) does no longer work for me. 

I tells me to login with a code that had been sent to another device but i do not have another device (ie i do not use a smartphone). 

After several failed attempts to log in somwhow i was prompted to install a new version and was warned that it has been signed by a different signee (john-preston) compared to the old one. I cannot remember exatly the steps that brought me here. 

Anyway i installed this new version after i found out John Preston is a developer involved wth the project. 

But the situation remains the same: when i open Telegram it asks me to login with a QR-Code (which i cannot use). Then i switch to 'login using your phone number' and enter my phone number. 

Then again i am prompted to look at another device. 

I am totally stuck, cannot communicate since two days and i have absolutely no idea how to get back into my chats. 

This has never happened before since i use telegram (approx. for 2 years).

Any hint or help very much appreciated, thanks to everybody, 
and have a nice day.

**Comment by Aokromes** [comment]
> Hello erveryone.
> Since yesterday Telegram desktop for mac (not sure about the version as i already tried to update telegram) does no longer work for me.
> 
> I tells me to login with a code that had been sent to another device but i do not have another device (ie i do not use a smartphone).
> 
> After several failed attempts to log in somwhow i was prompted to install a new version and was warned that it has been signed by a different signee (john-preston) compared to the old one. I cannot remember exatly the steps that brought me here.
> 
> Anyway i installed this new version after i found out John Preston is a developer involved wth the project.
> 
> But the situation remains the same: when i open Telegram it asks me to login with a QR-Code (which i cannot use). Then i switch to 'login using your phone number' and enter my phone number.
> 
> Then again i am prompted to look at another device.
> 
> I am totally stuck, cannot communicate since two days and i have absolutely no idea how to get back into my chats.
> 
> This has never happened before since i use telegram (approx. for 2 years).
> 
> Any hint or help very much appreciated, thanks to everybody,
> and have a nice day.

there are 2 different macos telegram.

Telegram for macos coded on swift
and 
Telegram desktop (called lite on mac store)

Maybe you had the 1st, and idk how you changed to the 2nd.
If you don't own atm a smartphone you may have to try https://github.com/kotatogram/kotatogram-desktop/releases to get login code then change to telegram desktop for updated version.

**Comment by ararename** [comment]
Hello there 

and thank you for your answer. 

I am sorry but i don't understand your answer. 

Ad i don't understand any of this: 

> you may have to try https://github.com/kotatogram/kotatogram-desktop/releases 
> to get login code then change to telegram desktop for updated version.

I had downloaded Telegram desktop for Mac about two years ago 
and always updated the App. 

Just yesterday this was first time Telegram ask me to log in in a new way. 
Never before. 

I am really stuck and i have no idea how the page you show me can help :(

Cheers 
Frank

**Comment by Aokromes** [comment]
> Hello there and thank you for your answer. I am sorry but i don't understand your answer. Ad i don't understand any of this:
> you may have to try [kotatogram/kotatogram-desktop/releases](https://github.com/kotatogram/kotatogram-desktop/releases) to get login code then change to telegram desktop for updated version.
> I had downloaded Telegram desktop for Mac about two years ago (from here: [macos.telegram.org](https://macos.telegram.org/)) and always updated the App. Just yesterday it was the first time that Telegram asked me to log in in a new way. Never before. Nothing changed on my system (afaik). I am really stuck and i have no idea how the page you showed me can help :( Cheers Frank

It's one alternative client, if you have a phone who is not smartphone you can get sms on it and get activation code there and put on it you can login on telegram again, then you change to the official client and put your phone on it and get activation code on alternative client, when you are loged in on official client you can delete kotatogram.

**Comment by ararename** [comment]
Got it :)

Thank you for your quick reply!! 

Will try and report back.

**Comment by ararename** [comment]
Uh, have to try tomorrow, have to go. 

Do you, or anyone else, by the way, know why this happened? 
As i said, fist time, never before in two years. 

I could not find any info that something has changed (ie the login procedure or something else).

**Comment by ararename** [comment]
Additional info: i do have 'two-step verification' active.

**Comment by john-preston** [comment]
This was a server side change, you can log in only using a smartphone with the Telegram app on it.

**Comment by ararename** [comment]
Ok, thank you for the answer. 

So you have effectively kicked out everyone who only uses the desktop app and no smarthphone. 

Congratulations....
Very customer friendly. 

Will check if this is the same with "Signal", and, 
sorry to say, but, bye, bye...

**Comment by john-preston** [comment]
@ararename Sorry, not my call.

**Comment by ararename** [comment]
Thanks for your reply. 
With Signal it is the same. 

Leaves room for speculations as i heard rumours that the App-Store-versions might be compromised and now people are being forced to use that version and not the one that is downloadable from the telegram.org servers (that's what i've been told, anyway). 

Do you know why that server side change was introduced? 
There might be a reason why you started to fence out desktop users. 
Or do i just not understand some crucial point here, maybe?

Do you, by chance, know any messenger that is still usable without a smartphone? 
Or am i, finally, forced to get me one?

Thanks again, cheers and have a nice day!

**Comment by john-preston** [comment]
@ararename In case of Telegram one can verify (not easily, but robustly) that AppStore versions (both iOS and Android) are not compromised, because they use reproducible builds and you can know for sure that the version in the store is built from exact public source code.

Also you can use Android version downloaded directly from https://telegram.org/android without AppStore.

Also you're not forced to use mobile version, you need it to log in first time on desktop app, then you can log out on your phone, uninstall it and never open again unless you loose your desktop session (you can log in new desktop sessions using the existing one without phone access).

As I understand in case of mobile apps it is somehow easier (a lot) to fight spam / abusive accounts that use modified software to sign up / log in thousands of bots (not the BotAPI-bots, but the bad ones, used for spam / incrementing fake views counters in channels / etc). It means almost nothing for the regular users except they indeed require iOS or Android smartphone (once, for the initial session log in).

**Comment by ararename** [comment]
This is very helpful, your information. Thank you very much indeed. 
It might take a while but in this case i will be back on the Telegram-train in the near future. 
Thanks again!!!!

(I did not know that Android gives me the freedom to not use a dedicated app store; it's not even that i need to hide my conversations but i have an overall tendency to not like proprietary corporate communication software :) 

And i will look into the verification option to verify builds of the software, also will tell my friends. 

Have a nice day.

**Comment by john-preston** [comment]
@ararename You can read about reproducible builds here:

https://core.telegram.org/reproducible-builds

The process of verification is complicated, but it is the best the team could come up with for the AppStore builds.

**Comment by ararename** [comment]
I am acually on a macbook but i will not buy into their ecosystem by also using an iphone. 
Will try the Android way, somehow, but need a phone first :)
